test-e2e/docker.ts

```diff
@@ -0,0 +1,124 @@
+import Docker from 'dockerode';
+import { scopedLogger } from '../src/lib/logger.js';
+
+const logger = scopedLogger('docker-manager');
+
+class DockerManager {
+	docker: Docker;
+
+	constructor () {
+		this.docker = new Docker();
+	}
+
+	public async createApiContainer () {
+		const isLinux = await this.isLinuxHost();
+		// docker run -e NODE_ENV=test -e TEST_MODE=e2e -e NEW_RELIC_ENABLED=false -e REDIS_URL=redis://host.docker.internal:6379 -e DB_CONNECTION_HOST=host.docker.internal --name globalping-api-e2e globalping-api-e2e
+		const container = await this.docker.createContainer({
+			Image: 'globalping-api-e2e',
+			name: 'globalping-api-e2e',
+			Env: [
+				'NODE_ENV=test',
+				'TEST_MODE=e2e',
+				'NEW_RELIC_ENABLED=false',
+				`REDIS_URL=redis://${isLinux ? 'localhost' : 'host.docker.internal'}:6379`,
+				`DB_CONNECTION_HOST=${isLinux ? 'localhost' : 'host.docker.internal'}`,
```

Comment:
Oh I see, probably https://github.com/jsdelivr/globalping/pull/504/commits/1a05f0858798c9509e556c0ad628888b7c4b97fb ? That bug is indeed annoying, but it still makes sense to have both versions of that command.

Reply:
This condition is related to docker behaviour. It is still `test` env in all cases, but:
- `--network host` is not working on mac/windows. https://github.com/docker/for-mac/issues/2716
- `gateway.docker.internal` is not working on linux.

I don't think this should be fixed using config, as that value should change based on the host OS.

Reply:
So if I understand correctly, right now, it uses the internal if we run it locally and host mode on CI?

My problem here is that the tests do not actually start redis on that port, they just assume there is one. It would be fine if the tests picked a random free port, started redis, and then shut it down. But if it's up to me to start it, then the number must be configurable. Default config can be for the CI and you can use the local config file to override for whatever your OS needs.

Reply:
> So if I understand correctly, right now, it uses the internal if we run it locally and host mode on CI?

Yes, but only because we use mac/windows locally, and linux on CI. Which may change or be different for other people. That condition is specific to OS, not to the running context (dev, test, ...).

I agree that hardcoded redis port should be removed, I can use value from config and conditionally do `redis.url.replace('localhost', 'docker.host.internal')` if that is required. Is that ok?

Reply:
Sure, that seems good.